feat: We love Liquid Glass

cantpr09ram opened the pull request:
This pull request refactors and improves the widget and view logic for course and student ID display, focusing on better handling of "not logged in" states, code simplification, and improved test coverage. The changes also update project file memberships for better modularity.

**Widget and View Improvements:**

* Unified and enhanced the "not logged in" UI across widgets (`CoursesNextUpSmallView`, `CoursesNextUpViewLockScreenView`, and `StdIDEntryView`) by adding a consistent icon and message, improving user feedback when SSO is missing. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL24-R62) [[2]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL16-R32) [[3]](diffhunk://#diff-b6434d7f4357be6e551eee496eca295f0f8608882fe47bb12d25e510187c9c64L58-R70)
* Improved the empty courses state with clearer messaging and visuals ("下週見") in both small and lock screen widgets. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL24-R62) [[2]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL16-R32)
* Refined course list rendering in `CoursesNextUpSmallView` for better clarity, including weekday name calculation, icon usage, and visual grouping of courses on the same day. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL52-R73) [[2]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fR124-R178) [[3]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL130-L136)

**Logic and Codebase Simplification:**

* Refactored `Provider` in `CoursesWidget.swift` to simplify weekday calculations, remove redundant helper methods, and use a service for next-up course logic. [[1]](diffhunk://#diff-3c6bc3911231e6cd7bed6025a5108c3ae48060c37482c14e8a343397e1c97adcL12-L66) [[2]](diffhunk://#diff-3c6bc3911231e6cd7bed6025a5108c3ae48060c37482c14e8a343397e1c97adcL100-L130)
* Updated previews and test data for more comprehensive widget state coverage, including logged out and edge-case scenarios. [[1]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL93-R113) [[2]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fR218-R242) [[3]](diffhunk://#diff-b6434d7f4357be6e551eee496eca295f0f8608882fe47bb12d25e510187c9c64R100-R101)

**Testing Enhancements:**

* Added a new test to ensure that on Sunday afternoons with no remaining courses, the widget displays the next Monday's course, improving reliability for weekly transitions.
* Clarified and adjusted the test for filtering out courses ending within 20 minutes, ensuring correct behavior.

**Project Structure:**

* Updated Xcode project file memberships to reflect the new modular structure, moving course-related utilities into a dedicated folder and ensuring correct test target inclusion. [[1]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838L157-R172) [[2]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838L183-R194) [[3]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838R216-R229)

cantpr09ram committed:
refactor: update course retrieval logic and enhance UI for login states

cantpr09ram committed:
refactor: clean up code structure and remove unused methods in various views

cantpr09ram committed:
refactor: clean up code structure and remove unused methods in various views

cantpr09ram committed:
feat: add EventEditSheet and EventDetailSheet for event editing and viewing

cantpr09ram committed:
Refactor course logic and modularize utilities

cantpr09ram committed:
Add campus map and landmark views, refactor course details

Introduced MapView and LandmarkView for campus mapping and location previews. Refactored CourseDetailView to show map based on room code and simplified CourseCardView and CourseView by removing ongoing status logic. Updated OtherView to link to the new MapView. Added Letter2Coordinate utility for mapping room codes to coordinates. Updated localizations for map-related actions.

cantpr09ram committed:
Add new app icon assets and update icon configuration

Introduces new SVG assets and icon.json files for both AppIcon and AppIconDev, including body, wing, eyes, and background components. Updates Xcode project to use AppIconDev for both Debug and Release builds and disables inclusion of all app icon assets.

cantpr09ram committed:
Refactor course schedule views and remove status overlays

Simplifies CourseScheduleView by removing floating cache update notifications and ongoing status indicators. Cleans up CourseCardView by eliminating time-based status logic and related UI, and removes the library button from DayScheduleView. Deletes CacheUpdateNotificationView and OngoingStatusIndicator components as they are no longer used.

cantpr09ram committed:
Improve next up course logic and widget weekday handling

Refines the logic for determining upcoming courses, including filtering out courses ending within 20 minutes and showing next week's Monday course if Sunday afternoon has no more classes. Updates widget weekday calculation to use ISO weekday, improves weekday name display, and enhances the small widget's course display. Adds a new test for Sunday afternoon behavior and updates related test cases. Also sets the 'glass' property in the app icon config.

cantpr09ram committed:
Replace WifiView with MapView in OtherView

Updated the .map case in OtherView to use MapView instead of WifiView, ensuring the correct view is displayed for the map option.

cantpr09ram committed:
Fix typo in coordinate for 視聽教育館

Corrected the longitude value for the 視聽教育館 (V) entry in letterLocations by adding a missing decimal point.

cantpr09ram committed:
Refactor week schedule and timeline UI for clarity

Simplifies and unifies weekday handling across WeekScheduleView and WeekdaysView, now always displaying all 7 days with correct date labels and highlighting for today. Refactors TimelineView to improve course card positioning, overlap, and vertical gap logic, and updates CourseView to remove ongoing status logic and use system background colors for better appearance. These changes improve maintainability and visual consistency of the weekly course schedule UI.

cantpr09ram committed:
Add appearance-based specializations to icon configs

cantpr09ram committed:
Remove Reachability from third-party libraries list

cantpr09ram committed:
Update calendar button style in EventView

Refines the 'Add to Calendar' button by using a Label with improved icon styling, increased padding, and a rounded background for better visual consistency and accessibility.

cantpr09ram committed:
Refine course card and week view UI styles

Adjusted spacing, font sizes, and padding in CourseCardView for a more compact layout. Updated CourseView to enhance badge styling, use more distinct colors, and improve text emphasis. Removed file header comments from WeekScheduleView for cleaner code.

cantpr09ram committed:
Add location list and enhance map marker selection

Refactored L2GData to include id and code properties and made it Identifiable. Added LocationListView for displaying selectable locations. Updated MapView to use the new LocationListView in a sheet, improved marker selection with animation, and updated MarkerSheetView to display the location code.

cantpr09ram committed:
v2.0.67

cantpr09ram committed:
Refactor code formatting and improve consistency

This commit applies consistent code formatting, indentation, and line wrapping across multiple Swift files, improving readability and maintainability. No functional changes were made; only whitespace, alignment, and minor stylistic adjustments were applied.

cantpr09ram commented:
hey @copilot can you provide right setting for running xcodebuild on GitHub Actions

cantpr09ram committed:
Update CI workflow, README, and Xcode project settings

Changed GitHub Actions workflow to use updated macOS runners and iOS Simulator destination. Cleaned up the README by removing feature descriptions and acknowledgments. Updated Xcode project file to add AppIconDev.icon and AppIcon.icon to build file exceptions.

cantpr09ram committed:
Update CI to use macos-latest runner

cantpr09ram committed:
Update ci.yml

winstonsung commented on .github/workflows/ci.yml:
```diff
@@ -44,7 +43,10 @@ jobs:
+            -configuration Debug \
+            -destination 'generic/platform=iOS Simulator' \
+            -sdk iphonesimulator \
             clean build
```
Undo removed newline at the end of the file.

winstonsung commented on README.md:
```diff
@@ -1,16 +1 @@
 # Dauphin
```
Is there any reason to blank the README?

winstonsung commented on App/dauphin.xcodeproj/project.pbxproj:
```diff
@@ -791,8 +819,9 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIconDev;
```
Intended?

winstonsung commented on App/dauphin.xcodeproj/project.pbxproj:
```diff
@@ -816,11 +845,11 @@
 				INFOPLIST_KEY_UISupportedInterfaceOrientations = UIInterfaceOrientationPortrait;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad = "UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown";
-				IPHONEOS_DEPLOYMENT_TARGET = 16.6;
+				IPHONEOS_DEPLOYMENT_TARGET = 17.6;
```
Just to ensure: there are changes that make the app unusable in iOS 16.6, right?

winstonsung replied:
?

winstonsung commented on App/dauphin/Localizable.xcstrings:
```diff
@@ -286,14 +289,12 @@
     "Enjoy your free day!" : {
 
     },
     "First" : {
```
Remove stale strings.

winstonsung commented on App/dauphin/Localizable.xcstrings:
```diff
@@ -594,6 +599,9 @@
     "Your student ID is %@" : {
 
+    },
+    "校務行事曆" : {
```
Fix source string: should in English.

winstonsung commented on App/dauphin/Utilities/Letter2Coordinate.swift:
```diff
@@ -0,0 +1,160 @@
+}
+
+let letterLocations: [String: L2GData] = [
+  "X": L2GData(
```
Location code "X" is actually located/assigned to "Five-tiger Hill Parking Lot".

Suggestion: location code "ZZZ" is preserved for "other location".

winstonsung commented on App/dauphin/Utilities/Letter2Coordinate.swift:
```diff
@@ -0,0 +1,160 @@
+  ),
+  "Q": L2GData(
+    code: "Q",
+    name: "教育館",
```
No longer "教育館", use "傳播Q館" instead.

winstonsung commented on App/dauphin/Utilities/Letter2Coordinate.swift:
```diff
@@ -0,0 +1,160 @@
+  ),
+  "O": L2GData(
+    code: "O",
+    name: "傳播館",
```
"傳播O館".

winstonsung commented on App/dauphin/View/CourseSchedule/Day/DayScheduleView.swift:
```diff
@@ -112,7 +51,7 @@ struct DayScheduleView: View {
               .foregroundColor(.secondary)
 
-            Text("No courses for \(dates[selectedDateIndex].weekday)")
+            Text("No courses for Today")
```
Format: "No courses for today" or "No Courses for Today".

winstonsung commented on App/dauphin/View/Other/Library/LibraryView.swift:
```diff
@@ -5,7 +5,6 @@
 //  Created by \u8b19 on 11/25/24.
 //
 
-import Code39
```
😂
(no further action required for this comment)

winstonsung commented on App/CoursesWidget/Views/NextUpLockScreenView.swift:
```diff
@@ -13,20 +7,32 @@ struct CoursesNextUpViewLockScreenView: View {
+        Image(systemName: "person.text.rectangle.trianglebadge.exclamationmark.fill")
+          .font(.system(size: 40, weight: .semibold))
+
+        Text("尚未登入")
```
Localizable?

cantpr09ram committed:
Move project files under App/

cantpr09ram committed:
Move project files under App/

cantpr09ram committed:
Move .swift-format to App directory and clean up project file

Relocated the .swift-format configuration file into the App directory. Removed obsolete PBXFileSystemSynchronizedBuildFileExceptionSet entries and related exceptions from the Xcode project file to simplify test target configuration.

cantpr09ram committed:
Add Docs/index.html landing page

Create a simple Docs/index.html as a project landing page for “Dauphin”. Includes embedded CSS and Google Bungee font, a responsive large headline, and two SVG icon links (App Store placeholder and GitHub repo). Adds basic centered layout and styling for icons and typography.

cantpr09ram committed:
test

cantpr09ram committed:
Add GitHub Actions workflow for Jekyll site build and artifact upload

cantpr09ram committed:
Rename Docs to Docs_tmp

cantpr09ram committed:
Rename Docs_tmp to docs

cantpr09ram committed:
Remove GitHub Actions workflow for Jekyll site and update README with CI and deployment badges

cantpr09ram commented:
@winstonsung 這個要多分幾次修，我先合併再一個一個處理可以嗎？
